lightning-payjoin/src/scheduler.rs

```diff
@@ -0,0 +1,378 @@
+use std::time::Instant;
+
+use bitcoin::{psbt::Psbt, secp256k1::PublicKey, ScriptBuf, Txid};
+use lightning::ln::{msgs::SocketAddress, ChannelId};
+
+/// A lightning network channel scheduler.
+///
+/// This struct is used to schedule and track the state of a channel creation process.
+#[derive(Clone)]
+pub struct ChannelScheduler {
+	channels: Vec<ScheduledChannel>,
+}
+
+impl ChannelScheduler {
+	/// Create a new empty channel scheduler.
+	pub fn new() -> Self {
+		Self { channels: vec![] }
+	}
+	/// Schedule a new channel.
+	///
+	/// The channel will be created with `ScheduledChannelState::ChannelCreated` state.
+	pub fn schedule(
+		&mut self, channel_value_satoshi: u64, push_msat: Option<u64>, announce_channel: bool,
+		node_id: PublicKey, user_channel_id: u128, address: SocketAddress,
+	) {
+		let channel = ScheduledChannel::new(
```

Comment:
It would, I thought in case of failure and a retry, we would want to have the data in hand but iam not sure its something we want to implement for now..